apps/ledger-live-desktop/src/renderer/analytics/hooks/useTrackTransactionChecksFlow.ts

```diff
@@ -0,0 +1,47 @@
+import { Device } from "@ledgerhq/types-devices";
+import { track } from "../segment";
+import { CONNECTION_TYPES } from "./variables";
+import { useRef } from "react";
+import { DeviceInfo } from "@ledgerhq/types-live";
+import { AppAndVersion } from "@ledgerhq/live-common/hw/connectApp";
+
+export type UseTrackTransactionChecksFlow = {
+  location: string | undefined;
+  device: Device;
+  deviceInfo: DeviceInfo | undefined | null;
+  appAndVersion: AppAndVersion | undefined | null;
+  transactionChecksOptInTriggered: boolean | undefined | null;
+  isTrackingEnabled: boolean;
+};
+
+export const useTrackTransactionChecksFlow = ({
+  location,
+  device,
+  deviceInfo,
+  appAndVersion,
+  transactionChecksOptInTriggered,
+  isTrackingEnabled,
+}: UseTrackTransactionChecksFlow) => {
+  const triggered = useRef(false);
+
+  if (transactionChecksOptInTriggered && !triggered.current) {
+    triggered.current = true;
+
+    const defaultPayload = {
+      page: location ?? "unknown",
+      deviceType: device?.modelId,
+      connectionType: device?.wired ? CONNECTION_TYPES.USB : CONNECTION_TYPES.BLE,
+      platform: "LLD",
+      appName: appAndVersion?.name,
+      appVersion: appAndVersion?.version,
+      appFlags: appAndVersion?.flags.toString(),
+      deviceInfoHardwareVersion: deviceInfo?.hardwareVersion,
+      deviceInfoMcuVersion: deviceInfo?.mcuVersion,
+      deviceInfoBootloaderVersion: deviceInfo?.bootloaderVersion,
+      deviceInfoProviderName: deviceInfo?.providerName,
+      deviceInfoLanguageId: deviceInfo?.languageId,
+    };
+
+    track("Transaction Check Opt-in Triggered", defaultPayload, isTrackingEnabled);
+  }
```

Comment:
Why do we prefer using a ref instead of a useEffect here ?

Reply:
this component does not need to re render as it's only tracking, also we need to ensure that the event is fired only once so `useRef` is working great in this situation

Reply:
Yes but it's just that in react you usually don't do side effects in a render phase and should use a `useEffect` with no dependencies for an side effect that needs to run only once